### PR TITLE
feat: Add docs for cross-region RPC

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/docs/backend/cross-region-rpc.mdx [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+title: "Cross Region RPC"
+---
+
+When sentry is deployed in a multi-region deployment (like sentry.io) there are many scenarios and workflows where a region silo requires information that is stored in Control Silo. Similarly there are flows where Control Silo operations need to read or mutate state stored in regions.
```
Comment (mrduncan) [inline src/docs/backend/cross-region-rpc.mdx:5]:
```suggestion
When Sentry is deployed in a multi-region deployment (like sentry.io) there are many scenarios and workflows where a region silo requires information that is stored in Control Silo. Similarly there are flows where Control Silo operations need to read or mutate state stored in regions.
```

File: src/docs/backend/cross-region-rpc.mdx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+When data needs to be read, or mutated synchronously, we use Remote Procedure Calls (RPC).
+
+## Design Context
+
+Before we get into specifics, it is useful to understand the requirements that the system was designed under. We needed the following attributes in our solution:
+
+- Needs to have consistent parameter and return types in local development, CI and multi-region production.
+- Serialization and transport concerns should be hidden from most application logic, as the transport depends on how the application is deployed.
+- Strong support for python typing
```
Comment (mrduncan) [inline src/docs/backend/cross-region-rpc.mdx:15]:
```suggestion
- Strong support for Python typing
```

File: src/docs/backend/cross-region-rpc.mdx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+### RPC Authentication
+
+All cross-region RPC requests go to a single endpoint `/api/0/internal/rpc/:service/:method` The RPC endpoint uses HMAC based signature comparison to validate authenticity. Before a request is made, the request body has an HMAC signature created using a secret that is shared by region and control silo instances.
+
+When an RPC message is received, the signature header is compared with a locally generated HMAC using the receiver’s secret. The request is only processed if the checksums match.
+
+### RPC serialization and transport
+
+Cross-Region RPC uses pydantic and openapi to provide typehinted parameter and response values, and fulfil our requirements for schema generation and validation. How RPC methods behave varies based on how the application is configured.
```
Comment (mrduncan) [inline src/docs/backend/cross-region-rpc.mdx:108]:
```suggestion
Cross-Region RPC uses pydantic and openapi to provide typehinted parameter and response values, and fulfill our requirements for schema generation and validation. How RPC methods behave varies based on how the application is configured.
```

File: src/docs/backend/cross-region-rpc.mdx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+- RPC methods are invoked synchronously on the implementation service.
+
+#### In tests
+
+- During tests we emulate as much of the RPC stack as we can.
+- Request bodies and responses are serialized as JSON.
+- Requests are made with `django.test.Client` to the RPC endpoint.
+
+#### In siloed mode
```
Comment (mrduncan) [inline src/docs/backend/cross-region-rpc.mdx:120]:
```suggestion
#### In siloed mode
```
Not sure if it's best to mention in the monolith mode section or here but might be worth mentioning the siloed mode case where the destination is local and no RPC request is actually made

**Review by mrduncan — APPROVED** [review]

File: src/docs/backend/cross-region-rpc.mdx (hunk 4 above)
Comment (markstory) [reply src/docs/backend/cross-region-rpc.mdx:120]: I think it works here - added.

**Pull request merged**